src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs

```diff
@@ -14319,6 +14319,154 @@ static void Test(params CollectionBuilderAttribute a)
                 );
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/74734")]
+        public void Cycle_16()
+        {
+            if (CompilationExtensions.EnableVerifyIOperation &&
+                ExecutionConditionUtil.Configuration == ExecutionConfiguration.Debug)
+            {
+                // A debug assert fails but an invalid operation is correctly created otherwise.
+                return;
+            }
+
+            var source = """
+                using System.Collections;
+                using System.Collections.Generic;
+
+                public class Base : IEnumerable<object>
```

Comment:
>Base [](http://example.com/codeflow?start=29&length=4)

Minor: All of the tests use a base and derived type. Is that necessary to repro the issue? If not, consider merging the base and derived in at least one test where there is currently only one derived type.

Reply:
Let's also test a generic collection type. Perhaps this test could be modified to:
```csharp
public class MyCollection1<T> : IEnumerable<T>
{
    public IEnumerator<T> GetEnumerator() => throw null;
    IEnumerator IEnumerable.GetEnumerator() => throw null;
    public void Add(params MyCollection1<T> c) {}
}

public class C
{
    MyCollection1<object> M() => [1];
}
```

Reply:
Consider testing:
```csharp
public class MyCollection1<T> : IEnumerable<T>
{
    public IEnumerator<T> GetEnumerator() => throw null;
    IEnumerator IEnumerable.GetEnumerator() => throw null;
    public void Add(params MyCollection1<object> c) {}
}

public class C
{
    MyCollection1<int> M() => [1];
}
```

Reply:
Should we test overloads, such as the following, which looks like it should succeed:
```csharp
public class MyCollection1<T> : IEnumerable<T>
{
    public IEnumerator<T> GetEnumerator() => throw null;
    IEnumerator IEnumerable.GetEnumerator() => throw null;
    public void Add(T x) {}
    public void Add(params MyCollection1<T> y) {}
}

public class C
{
    MyCollection1<object> M() => [1];
}
```